File: python/2022/day6/main.py
# Part 1
print((lambda x :[i for i in range(4, len(x)) if len(set(x[i - 4 : i])) == 4][0])(open("python/2022/day6/test.txt").readline()))
# Part 2
print((lambda x :[i for i in range(14, len(x)) if len(set(x[i - 14 : i])) == 14][0])(open("python/2022/day6/test.txt").readline()))

File: typescript/2022/day6/main.ts
import * as fs from "fs";

const data = fs
  .readFileSync("typescript/2022/day6/test.txt", "utf-8")
  .split("");

for (let i = 0; i < data.length; i++) {
  const set: Set<string> = new Set(data.slice(i, i + 4));
  if (set.size === 4) {
    console.log(i + 4);
    break;
  }
}
